src/server/services/wms/qgswmsgetlegendgraphics.cpp

```diff
@@ -284,10 +285,11 @@ namespace QgsWms
             legendNode->setUserLabel( QStringLiteral( " " ) );
           }
         }
-        else if ( !parameters.layerTitleAsBool() && model->layerLegendNodes( nodeLayer, true ).count() )
+        else if ( ruleLabel.compare( QStringLiteral( "AUTO" ), Qt::CaseInsensitive ) == 0 )
```

Comment:
Does this work when `LAYERTITLE=true&RULELABEL=auto` like it should or does it remove the rule labels as well? Shouldn't it check the LAYERTITLE parameter, like `!parameters.layerTitleAsBool() && ( model->layerLegendNodes( nodeLayer, true ).count() || ruleLabel.compare( QStringLiteral( "AUTO" ), Qt::CaseInsensitive ) == 0  ) )`?

Reply:
Oh. Sorry. I haven't received @m-kuhn comment on writing this one. If you care about his input @uprel you can ignore this one...

Reply:
OK, I answered above.